### PR TITLE
Set flag for managing the hostsfile

### PR DESCRIPTION
**Pull request by vervas**
Make hostsfile changes optional in case of manual handling of it in
other places.

### CODE REVIEW

**Comment by mburns** [comment]
I'm ok with it. :+1:

**Comment by mburns** [comment]
@vervas - could you rebase this PR so we can merge? Thanks!

**Comment by vervas** [comment]
Sorry for the late response. Rebasing is done.

File: recipes/default.rb [hunk 1]
```diff
@@ -100,19 +100,21 @@
     end
   end
 
-  hostsfile_entry 'localhost' do
-    ip_address '127.0.0.1'
-    hostname 'localhost'
-    action :append
-  end
+  if node['hostname_cookbook']['manage_hostfile']
```
Comment (shortdudey123) [inline recipes/default.rb]: `only_if` is a better option instead of enclosing the resource in an `if` block

Comment (vervas) [inline recipes/default.rb]: I'm happy to change this if there is a plan to merge and release

Comment (mburns) [inline recipes/default.rb]: I've [cut a new release](https://github.com/3ofcoins/chef-cookbook-hostname/releases/tag/v0.4.1), and would be happy to include this PR in another release if you wouldn't mind applying the requested changes.

File: recipes/default.rb [hunk 2]
```diff
@@ -100,19 +100,21 @@
...
-    action :create
-    notifies :reload, 'ohai[reload_hostname]', :immediately
+    hostsfile_entry 'set hostname' do
+      ip_address node['hostname_cookbook']['hostsfile_ip']
+      hostname fqdn
+      aliases aliases
+      unique true
+      action :create
+      notifies :reload, 'ohai[reload_hostname]', :immediately
+    end
```
Comment (mburns) [inline recipes/default.rb]: #52 does somewhat similar stuff, fwiw.

**Comment by vervas** [comment]
I suppose this can be closed since #52 is merged, right?

**Pull request closed without merging**